aminator/plugins/provisioner/aptitude.py

```diff
@@ -59,47 +64,59 @@ class AptitudeProvisionerPlugin(AptProvisionerPlugin):
 
     # overload this method to call aptitude instead.
     @classmethod
-    def apt_get_localinstall(cls, package):
-        """install deb file with dpkg then resolve dependencies
-        """
-        dpkg_ret = cls.dpkg_install(package)
+    def _fix_localinstall_deps(cls, package):
+        # use aptitude and its solver to resolve dependencies after a dpkg -i
         pkgname, _, _ = basename(package).split('_')
-        if not dpkg_ret.success:
-            # figure out the version via dpkg rather than parsing it out of the file name
-            # in case there is an epoch in the version
-            query_ret = super(AptitudeProvisionerPlugin,cls).deb_query(package, "${Version}", local=True)
-            if not query_ret.success:
-                log.debug('failure:{0.command} :{0.std_err}'.format(query_ret.result))
-            pkgver = query_ret.result.std_out
-
-            log.debug('failure:{0.command} :{0.std_err}'.format(dpkg_ret.result))
-            aptitude_ret = cls.aptitude("install", "{}={}".format(pkgname, pkgver))
-            if not aptitude_ret.success:
-                log.debug('failure:{0.command} :{0.std_err}'.format(aptitude_ret.result))
-            query_ret = super(AptitudeProvisionerPlugin,cls).deb_query(pkgname, "${Status} ${Version}", local=False)
-            if not query_ret.success:
-                log.debug('failure:{0.command} :{0.std_err}'.format(query_ret.result))
-            if "install ok installed" not in query_ret.result.std_out:
-                raise RuntimeError("package {} failed to be installed.  dpkg status: {}".format(pkgname, query_ret.result.std_out))
-            installed_version = query_ret.result.std_out.split()[-1].strip()
-            if installed_version != pkgver:
-                raise RuntimeError("package {} failed to be installed. requested version {}, aptitude installed {}".format(pkgname, pkgver, installed_version))
-            return query_ret
-        return dpkg_ret
+
+        # figure out the version via dpkg rather than parsing it out of the file name
+        # in case there is an epoch in the version
+        version_query_ret = cls.deb_query(package, "${Version}", local=True)
+        if not version_query_ret.success:
+            log.critical("Unable to query version for package {0}".format(package))
+            return version_query_ret
+        pkgver = version_query_ret.result.std_out
```

Comment:
Previously we'd just continue on even if these steps failed for some reason, so I returned on the critical failures so we stop early rather than try to do more crap when things are already broken